packages/algoliasearch/src/ingestion.ts

```diff
@@ -0,0 +1,167 @@
+import { AuthMode, ClientTransporterOptions, createAuth, encode } from '@algolia/client-common';
+import {
+  BatchActionEnum,
+  ChunkOptions,
+  PartialUpdateObjectsOptions,
+  SaveObjectsOptions,
+  SearchClientOptions,
+} from '@algolia/client-search';
+import { MethodEnum } from '@algolia/requester-common';
+import { CallEnum, createTransporter, RequestOptions } from '@algolia/transporter';
+
+import {
+  IngestionClient,
+  PushOptions,
+  PushProps,
+  TransformationOptions,
+  WatchResponse,
+} from './types';
+
+export function createIngestionClient(
+  options: SearchClientOptions & ClientTransporterOptions & TransformationOptions
+): IngestionClient {
+  if (!options || !options.transformation || !options.transformation.region) {
+    throw new Error('`region` must be provided when leveraging the transformation pipeline');
+  }
+
+  const appId = options.appId;
+
+  const auth = createAuth(
+    options.authMode !== undefined ? options.authMode : AuthMode.WithinHeaders,
+    appId,
+    options.apiKey
+  );
+
+  const transporter = createTransporter({
+    hosts: [
+      {
+        url: `data.${options.transformation.region}.algolia.com`,
+        accept: CallEnum.ReadWrite,
+        protocol: 'https',
+      },
+    ],
+    ...options,
+    headers: {
+      ...auth.headers(),
+      ...{ 'content-type': 'application/x-www-form-urlencoded' },
```

Comment:
is this volontary ? we only support application/json

Reply:
ah ty I forgot to match what we had in the generated clients here